### ISSUE TITLE
V5.0.1 Database migration Alert during upgrade

### ISSUE THREAD

**Issue by hixair** [grokability/snipe-it#8572]
#### Please confirm you have done the following before posting your bug report:

- [NA] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**

Following #8553 I was able to upgrade (by trashing the vendor folder, editing the .env file for the REDIS_PORT operator and updating my php modules)
I have a working SnipeIT instance, however I had the following alert related to the database migration during upgrade (using the 5.0.1 upgrade.php) : 

--------------------------------------------------------
Step 7: Migrating database:
--------------------------------------------------------

-- Migrating: 2018_07_28_023826_create_checkout_acceptances_table
Migrated:  2018_07_28_023826_create_checkout_acceptances_table (0.16 seconds)
Migrating: 2018_08_20_204842_add_depreciation_option_to_settings

In Connection.php line 669:
                                                                                       
  SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The max  
  imum row size for the used table type, not counting BLOBs, is 8126. This includes s  
  torage overhead, check the manual. You have to change some columns to TEXT or BLOBs  
   (SQL: alter table `settings` add `depreciation_method` char(10) null default 'defa  
  ult')                                                                                
                                                                                       

In PDOStatement.php line 129:
                                                                                       
  SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The max  
  imum row size for the used table type, not counting BLOBs, is 8126. This includes s  
  torage overhead, check the manual. You have to change some columns to TEXT or BLOBs  
                                                                                       

In PDOStatement.php line 127:
                                                                                       
  SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The max  
  imum row size for the used table type, not counting BLOBs, is 8126. This includes s  
  torage overhead, check the manual. You have to change some columns to TEXT or BLOBs  
                                                                                       

**To Reproduce**
I reverted to a working 4.9.5, pulled changes from git, and upgraded using the upgrade.php 

**Server (please complete the following information):**
 - Snipe-IT Version 4.9.5 to 5.0.1
 - OS: 4.19.0-11-amd64 #1 SMP Debian 4.19.146-1 (2020-09-17) x86_64 GNU/Linux
 - Web Server: Apache
 - PHP Version 7.3.19-1~deb10u1
 - Laravel Version 6.18.10

**Desktop (please complete the following information):**

**Comment by hixair** [comment]
Update : I get 500 errors whenever i checkout an asset, or try to save a new logo in the branding part of the settings.
I will revert back to 4.9.5 :'(

**Comment by snipe** [comment]
Congrats! I wrote a book on MySQL and I have NO IDEA what’s happening here, so that’s new and fun. 

Re: the 500 errors, that wouldn’t shock me. If migrations can’t complete, you’re gonna have a bad time. Can you give me more info on your version MySQL maybe?

**Comment by hixair** [comment]
I am running mysql  Ver 15.1 Distrib 10.3.23-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
I am also strongly considering moving to your hosted Snipe-IT 😆

**Comment by ISNL-PimP** [comment]
Got a similar issue after upgrading 4.9.5 to 5.0 with the upgrade.php script, got the same SQLSTATE errors about incorrect blob size.
I fixed it by running some SQL commands similar to what's described here: https://snipe-it.readme.io/docs/installation-issues#sqlstatehy000-general-error-1005-cant-create-table-snipeitsql-3626_1c6-errno-150

```SQL
USE snipeit;
 
ALTER TABLE companies ENGINE = InnoDB;
ALTER TABLE settings ENGINE=InnoDB;
```

After that, I reran the database migrations (`php artisan migrate`) which then completed without errors.

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/installation-issues#sqlstatehy000-general-error-1005-cant-create-table-snipeitsql-3626_1c6-errno-150">Installation/Upgrading Issues</a></strong></div><div>Common installation/upgrading issues and solutions</div></blockquote>

**Comment by snipe** [comment]
Do either/both of you have very old Snipe-IT installs? We've specifically stated InnoDB as the default for a few years now, so you shouldn't have to state that specifically. (Also InnoDB has bee the default engine for a while, but if you're running something that was installed a long time ago, it's possible we didn't make a migration specifically to handle that conversion.)

**Comment by snipe** [comment]
@hixair I promise we don't make this difficult on purpose. :D 

A lot of it is the ecosystem itself. Software (web servers, database, PHP, composer libraries, etc) all change, and while we do try to test everywhere we can, it would obviously be impossible for us to test *every* possible scenario. We want you to be successful whether you're hosted with us or not ;)

**Comment by hixair** [comment]
Hi !

Following all the help I got here, I got our instance working and the problem was indeed coming from the db format (that I never checked during upgrades).

Don’t worry, I am not complaining !
I have been running snipe-it since version 3 or earlier and it is far superior to every asset utility I ever tested or used ! 
I really liked that tool, I made a lot of automations for our device management with it and even gave a presentation at a conference about it. 

Actually, quickly after deploying our first instance, we figured that we will be testing snipe-it extensively and go to your hosted version. 
I had trouble having this cost validated on as it was not « business critical » but today **I finally did** (like a lot of companies, we are hosting fewer and fewer services on premises) !

Can’t wait to migrate our data (already contacted your support, this was pretty straightforward). 

I know you are facing a lot of complaints and criticism, but people quickly forget they are using a free product that is made by humans 😂

**Comment by snipe** [comment]
Thanks so much for the follow up! <del>I'll get this added this to the docs just in case anyone else runs into it as well. </del>

Derp, I had already added it to the docs :D 
https://snipe-it.readme.io/docs/installation-issues#during-database-migrations-max-key-length-is-1000-bytes

I just updated the description header to be a little clearer. I wonder if we should just add that to the upgrade.php script. It would execute every time that upgrade script was run, but that shouldn't hurt anything if the table type is already InnoDB to run that alter statement again. 

> I have been running snipe-it since version 3 or earlier and it is far superior to every asset utility I ever tested or used !

Wow, that's a wonderful endorsement, thank you! 

> I really liked that tool, I made a lot of automations for our device management with it and even gave a presentation at a conference about it.

If it was recorded, I'd love to see that presentation!

> I know you are facing a lot of complaints and criticism, but people quickly forget they are using a free product that is made by humans 

I mean, it's sort of expected tbh. This was a big update. There were some new PHP extensions required, and any time you do a major framework upgrade there's bound to be some bits that are haunted. Stuff that got cached that shouldn't have been cached, etc. It's been a bumpy road, but I'm so glad to be here finally. There's so much more we want to do that we couldn't until this hurdle had been crossed. I try not to take it too personally, and I genuinely do appreciate the fact that 1) so many people jumped on the upgrade already and 2) the time and care they've put into the bug reports. 

Closing this for now, since it looks like everybody's issues are fixed. :)

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/installation-issues#during-database-migrations-max-key-length-is-1000-bytes">Installation/Upgrading Issues</a></strong></div><div>Common installation/upgrading issues and solutions</div></blockquote>
